x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/datafeeds/RestPutDatafeedAction.java

```diff
@@ -44,8 +46,15 @@ public String getName() {
     @Override
     protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient client) throws IOException {
         String datafeedId = restRequest.param(DatafeedConfig.ID.getPreferredName());
+        IndicesOptions indicesOptions = null;
+        if (restRequest.hasParam("expand_wildcards")
```

Comment:
Could we avoid checking if `restRequest` has those params? I think it'd work to do `IndicesOptions indicesOptions = IndicesOptions.fromRequest(restRequest, SearchRequest.DEFAULT_INDICES_OPTIONS)`

Reply:
Makes sense. DatafeedConfig can make the options non-null and default to `SearchRequestt.DEFAULT_INDICES_OPTIONS`